base/loading.jl

```diff
@@ -163,39 +163,90 @@ function version_slug(uuid::UUID, sha1::SHA1, p::Int=5)
     return slug(crc, p)
 end
 
-struct TOMLCache
-    p::TOML.Parser
-    d::Dict{String, Dict{String, Any}}
+mutable struct CachedTOMLDict
+    path::String
+    inode::UInt64
+    mtime::Float64
+    size::Int64
+    hash::UInt32
+    d::Dict{String, Any}
+end
+
+function CachedTOMLDict(p::TOML.Parser, path::String)
+    s = stat(path)
+    content = read(path)
+    crc32 = _crc32c(content)
+    TOML.reinit!(p, String(content); filepath=path)
+    d = TOML.parse(p)
+    return CachedTOMLDict(
+        path,
+        s.inode,
+        s.mtime,
+        s.size,
+        crc32,
+        d,
+   )
+end
+
+function get_updated_dict(p::TOML.Parser, f::CachedTOMLDict)
+    s = stat(f.path)
+    time_since_cached = time() - f.mtime
+    if time_since_cached < 0.1 || s.inode != f.inode || s.mtime != f.mtime || f.size != s.size
```

Comment:
I don't understand the time portion of this...why reparse specifically when the file is fresh? What happens if another process rewrites the file?

Reply:
Because `mtime` has a finite resolution we have to be a bit careful if we just cached it because in that case it could get updated without us noticing.

Reply:
But maybe we shouldn't cache it in this case... Not sure what is best.

Reply:
> why reparse specifically when the file is fresh

Also, note that we only reparse if the hash ends up being different. Otherwise it's just the cost of reading the file.

Reply:
I added a comment and named the magical constant.